controllers/api/interestRoutes.js

```diff
@@ -25,12 +29,35 @@ router.get('/', async (req, res) => {
 
 // Input:   idea_id, user_id
 // Creates a new interest relation between user and idea
-router.post('/', async (req, res) => {
+router.post('/', withAuthJson, async (req, res) => {
   try {
-    await Interest.create(req.body);
-    res.status(200).json({
-      message: `Interest created`,
+    // only add interest if it doesn't already exist
+    const existingInterest = await Interest.findAll({
+      where: [
+        {
+          idea_id: req.body.idea_id,
+          user_id: req.session.user_id,
+        },
+      ],
     });
+
+    console.log(existingInterest);
+    if (!existingInterest.length) {
+      // Create interest if it doesn't exist
+      await Interest.create({
+        idea_id: req.body.idea_id,
+        user_id: req.session.user_id,
+      });
+      res.status(200).json({ interested: true });
+    } else {
+      // else remove interest
+      await Interest.destroy({
```

Comment:
Hey, @jmichaelbrown8, We shouldn't have a POST route doing both CREATE and DELETE actions. That would be a rather unintuitive experience for developers without personal knowledge of your code base.

I'm assuming this is in effort to create some sort of toggle feature? The "toggling" decision should be done in the client side JS. So if the user has already marked the Idea with "interest", it will make a DELETE request instead of a POST.